examples/elect_leaders.c

```diff
@@ -0,0 +1,341 @@
+
+        err = rd_kafka_ElectLeadersResult_error(res);
+
+        results = rd_kafka_ElectLeadersResult_partitions(res, &results_cnt);
+
+        if (err) {
+                printf("%% ElectLeaders failed: %s\n", rd_kafka_err2str(err));
+                retval = 1;
+        } else {
+                for (i = 0; i < results_cnt; i++) {
+                        if (rd_kafka_topic_partition_result_error(results[i])) {
+                                printf(
+                                    "%% ElectLeaders failed for %s [%" PRId32
+                                    "] : %s\n",
+                                    rd_kafka_topic_partition_result_topic(
+                                        results[i]),
+                                    rd_kafka_topic_partition_result_partition(
+                                        results[i]),
+                                    rd_kafka_topic_partition_result_error_string(
+                                        results[i]));
+                        } else {
+                                printf(
+                                    "%% ElectLeaders succeeded for %s [%" PRId32
+                                    "]\n",
+                                    rd_kafka_topic_partition_result_topic(
+                                        results[i]),
+                                    rd_kafka_topic_partition_result_partition(
+                                        results[i]));
+                        }
+                }
+        }
+
+        return retval;
+}
+
+/**
+ * @brief Parse an integer or fail.
+ */
+int64_t parse_int(const char *what, const char *str) {
+        char *end;
+        unsigned long n = strtoull(str, &end, 0);
+
+        if (end != str + strlen(str)) {
+                fprintf(stderr, "%% Invalid input for %s: %s: not an integer\n",
+                        what, str);
+                exit(1);
+        }
+
+        return (int64_t)n;
+}
+
+static void cmd_elect_leaders(rd_kafka_conf_t *conf, int argc, char **argv) {
+        rd_kafka_t *rk;
+        char errstr[512];
+        rd_kafka_AdminOptions_t *options;
+        rd_kafka_event_t *event = NULL;
+        rd_kafka_topic_partition_list_t *partitions;
+        rd_kafka_ElectionType_t election_type;
+        rd_kafka_ElectLeaders_t *elect_leaders;
+        int i;
+        int retval = 0;
+
+        if (argc < 3) {
+                usage("Invalid number of arguments");
+        }
+        int etype = parse_int("election_type", argv[0]);
+
+        if (etype == 0) {
+                election_type = RD_KAFKA_ELECTION_TYPE_PREFERRED;
+        } else if (etype == 1) {
+                election_type = RD_KAFKA_ELECTION_TYPE_UNCLEAN;
+        } else {
+                usage("Invalid election type");
+        }
+        argc--;
+        argv++;
+        partitions = rd_kafka_topic_partition_list_new(argc / 2);
+        if (argc % 2 != 0) {
+                usage("Invalid number of arguments");
+        }
```

Comment:
You should do this invalid number of argument check before itself when checking for less than 3 arguments. Check for 2n+1 arguments there.